### PR TITLE
ProcedureConfiguration API to support types known at runtime

### PR DESCRIPTION
**Pull request by YogiBear52**
- ProcedureConfiguration - Add a parameter by the given clrType (not only with the generic API)
- FunctionConfiguration/ActionConfiguration - Set a return type with the given clrType(not only with the generic API)

### CODE REVIEW

**Comment by xuzhg** [comment]
@YogiBear52  Would you please give us a scenario description why these functions (APIs) are necessary for you? Thanks.

**Comment by YogiBear52** [comment]
Sure,
I am trying to build and EdmModel out of a DataModel which is known only at runtime.
Entities,entity sets, functions and complex types.
These days I'm writing code in OData.WebApi and ODataLibs to help customize and be build all odata endpoint functonality for data models known at runtime.

The current API of function configuration is the generic method AddParameter<T>. Since I don't know the type T at compile time, I can't use this method (unless I use reflection, but it is not a good coding..)

Thanks.

File: OData/src/System.Web.OData/OData/Builder/ActionConfiguration.cs [hunk 1]
```diff
@@ -133,22 +133,36 @@ public ActionConfiguration HasActionLink(Func<EntityInstanceContext, Uri> action
         /// Established the return type of the Action.
         /// <remarks>Used when the return type is a single Primitive or ComplexType.</remarks>
         /// </summary>
-        [SuppressMessage("Microsoft.Design", "CA1004:GenericMethodsShouldProvideTypeParameter", Justification = "In keeping with rest of API")]
-        public ActionConfiguration Returns<TReturnType>()
+        public ActionConfiguration Returns(Type clrReturnType)
```
Comment (VikingsFan) [inline OData/src/System.Web.OData/OData/Builder/ActionConfiguration.cs]: can you add test case for this new API, same for the others

**Comment by xuzhg** [comment]
Merged. Thanks.

**Pull request closed without merging**